SweetTweetApp/src/main/java/posts/PostsFunctions.java

```diff
@@ -0,0 +1,126 @@
+package posts;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+public class PostsFunctions {
+
+    private static HashMap<String, Post> posts = new HashMap<>();
+
+    public PostsFunctions(){ }
+
+    public List<Post> getAllPosts(){
+        return new ArrayList<>(posts.values());
+    }
+
+    public boolean hasPost(String postId){
+        return posts.containsKey(postId);
+    }
+
+    public List<Post> getPosts(Integer skip, Integer top, PostFiltering filters) {
+        if (skip == null) {
+            skip = 0;
+        }
+
+        if (top == null) {
+            top = 10;
+        }
+
+        sortByDateDescending();
+
+        Collection<Post> collectionPosts=posts.values();
+
+        if (filters == null) {
+            return collectionPosts.stream().skip(skip).limit(top).collect(Collectors.toList());
+        }
+
+        return collectionPosts.stream().filter(filters::filtering).skip(skip).limit(top).collect(Collectors.toList());
+    }
+
+    public Optional<Post> getPost(String postId){
+        if(Integer.parseInt(postId)>0){
+            return posts.values().stream().filter(i -> i.getId().equals(postId)).findFirst();
+        }
+        else{
+            return null;
+        }
+    }
+
+    public static boolean validatePost(Post post){
+        try {
+            if(Integer.parseInt(post.getId())<0){
```

Comment:
Spaces, you remember)
Also, the multitude of this checkings looks... a bit naive, In a commercial project, they would definitely do it other way. There are different ways. Here is one:
```
return Stream.of(post.getCreatedAt(), post.getAuthor(), post.getDescription() ...).allMatch(Objects::nonNull) && Integer.parseInt(post.getId()) >= 0;
```
Yet this is not a mistake, just a code style notion.

Btw, why not making Post.id `int` from the start - to avoid this naughty type coersion?